src/shared_memory_manager.cc

```diff
@@ -264,6 +264,46 @@ OpenCudaIPCRegion(
 
   return nullptr;
 }
+
+TRITONSERVER_Error*
+GetCudaSharedMemoryRegionSize(CUdeviceptr data_ptr, size_t& shm_region_size)
+{
+  CUdeviceptr* base = nullptr;
+  CUresult result = cuMemGetAddressRange(base, &shm_region_size, data_ptr);
+  if (result != CUDA_SUCCESS) {
+    const char* errorString;
+    if (cuGetErrorString(result, &errorString) != CUDA_SUCCESS) {
+      return TRITONSERVER_ErrorNew(
+          TRITONSERVER_ERROR_INTERNAL, "Failed to get CUDA error string");
+    }
+    return TRITONSERVER_ErrorNew(
+        TRITONSERVER_ERROR_INTERNAL,
+        std::string(
+            "Failed to get CUDA address range: " + std::string(errorString))
+            .c_str());
+  }
+  return nullptr;
+}
+
+TRITONSERVER_Error*
+CheckCudaSharedMemoryRegionSize(
+    const std::string& name, CUdeviceptr data_ptr, size_t byte_size)
+{
+  size_t shm_region_size = 0;
+  RETURN_IF_ERR(GetCudaSharedMemoryRegionSize(data_ptr, shm_region_size));
+
+  // User-provided offset and byte_size should not go out-of-bounds.
+  if (byte_size > shm_region_size) {
+    return TRITONSERVER_ErrorNew(
+        TRITONSERVER_ERROR_INVALID_ARG,
+        std::string(
+            "failed to register CUDA shared memory region '" + name +
+            "': register size exceeds CUDA shared memory region size")
```

Comment:
In system shm equivalent, I was guided to keep the error messages more vague: https://github.com/triton-inference-server/server/blob/96e4c0445d1329ee006fac6c58fdb39cbbf3e8a0/src/shared_memory_manager.cc#L187-L188.

@nnshah1 asuming you'd like the same here.

Reply:
this is on the border - but I think keeping the reasoning as clear but not leaking details is better. 

"invalid size" could be an alternate - but lets' match / use the same messaging as in system shared memory

Reply:
Sounds good. Updated the error message.